### PR TITLE
Fix clientActivityId being null. It is not guaranteed to be present for all channels.

### PR DESCRIPTION
**Pull request by MattSFT**
If client activity id is null then just use generic 'messageId' as the id for the message.
Add tests to ensure we don't regress.

### CODE REVIEW

File: src/adapter-middleware.coffee [hunk 1]
```diff
@@ -28,7 +28,10 @@ class TextMiddleware extends BaseMiddleware
         user.activity = activity
 
         if activity.type == 'message'
-            return new TextMessage(user, activity.text, activity.sourceEvent.clientActivityId)
+            messageId = activity.sourceEvent?.clientActivityId
```
Comment (maxpert) [inline src/adapter-middleware.coffee]: Will `return new TextMessage(user, activity.text, activity.sourceEvent?.clientActivityId || '')` work?

File: src/adapter-middleware.coffee [hunk 2]
```diff
@@ -28,7 +28,10 @@ class TextMiddleware extends BaseMiddleware
...
+            if not messageId?
+                messageId = 'message-id'
```
Comment (maxpert) [inline src/adapter-middleware.coffee]: Not sure if `message-id` buys us anything can it be empty string?

File: src/adapter-middleware.coffee (hunk 1 above)
Comment (MattSFT) [reply src/adapter-middleware.coffee]: Yes that's fine

File: src/adapter-middleware.coffee (hunk 2 above)
Comment (MattSFT) [reply src/adapter-middleware.coffee]: I'm not sure. I saw the shell adapter just throws message id in there. I think any string will work fine

**Pull request merged**